.circleci/config.yml

```diff
@@ -4,98 +4,53 @@
 #
 version: 2
 references:
-  container_python36: &container_python36
+  container_python310: &container_python310
     docker:
-      - image: circleci/python:3.6.2
+      - image: cimg/python:3.10.11
       - name: localstack
-        image: localstack/localstack:0.11.5
+        image: localstack/localstack:2.0.2
     working_directory: ~/repo
 
-  restore_cache_python36: &restore_cache_python36
+  restore_cache_python310: &restore_cache_python310
     restore_cache:
       keys:
         - v1-dependencies-{{ checksum "requirements.txt" }}-{{ checksum "requirements-dev.txt" }}
         # fallback to using the latest cache if no exact match is found
         - v1-dependencies-
 
-  save_cache_python36: &save_cache_python36
+  save_cache_python310: &save_cache_python310
     save_cache:
       paths:
-        - ~/venv36
-      key: v1-dependencies-{{ checksum "requirements.txt" }}-{{ checksum "requirements-dev.txt" }}
-
-
-  container_python38: &container_python38
-    docker:
-      - image: circleci/python:3.8.1
-      - name: localstack
-        image: localstack/localstack:0.11.5
-    working_directory: ~/repo
-
-  restore_cache_python38: &restore_cache_python38
-    restore_cache:
-      keys:
-        - v1-dependencies-{{ checksum "requirements.txt" }}-{{ checksum "requirements-dev.txt" }}
-        # fallback to using the latest cache if no exact match is found
-        - v1-dependencies-
-
-  save_cache_python38: &save_cache_python38
-    save_cache:
-      paths:
-        - ~/venv38
+        - ~/venv310
       key: v1-dependencies-{{ checksum "requirements.txt" }}-{{ checksum "requirements-dev.txt" }}
 
 
 jobs:
-  build_python36:
-    <<: *container_python36
-    steps:
-      - checkout
-      - *restore_cache_python36
-      - run:
-          name: install dependencies
-          command: |
-            pip install --user virtualenv
-            ~/.local/bin/virtualenv ~/venv36
-            . ~/venv36/bin/activate
-            pip install -r requirements.txt
-            pip install -r requirements-dev.txt
-            pip install .
-      - *save_cache_python36
-      - run:
-          name: run tests
-          environment:
-            LOCALSTACK_HOST: localstack
-          command: |
-            . ~/venv36/bin/activate
-            pylint_runner
-            CUMULUS_ENV=testing nosetests -v -s
-
-  build_python38:
-    <<: *container_python38
+  build_python310:
+    <<: *container_python310
     steps:
       - checkout
-      - *restore_cache_python38
+      - *restore_cache_python310
```

Comment:
Note that we're removing the 3.8 build/test here and ONLY testing for 3.10. I'm doing this because our Core Lambdas will run 3.10 so supporting earlier versions didn't seem to make sense. I'm very open to pushback though as I'm not 100% sure this is the right call.